### ISSUE TITLE
Preventing `oejut.QueuedThreadPool:qtp1928973631-184: Job failed` on abruptly-ended requests

### ISSUE THREAD

**Issue by dutchmahoney** [eclipse/jetty.project#9178]
**Jetty version**

 `11.0.13`

**Java version**

 `17.0.4.1`

**Background**

I'm using Jetty's async capabilities and seeing a lot of the following in my logs:

```
2023-01-16 20:40:38.611:WARN :oejut.QueuedThreadPool:qtp1928973631-184: Job failed
java.lang.IllegalStateException: ABORTED
        at org.eclipse.jetty.server.HttpChannelState.sendError(HttpChannelState.java:911)
        at org.eclipse.jetty.server.HttpChannelState.nextAction(HttpChannelState.java:493)
        at org.eclipse.jetty.server.HttpChannelState.unhandle(HttpChannelState.java:418)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:659)
        at org.eclipse.jetty.server.HttpChannel.run(HttpChannel.java:457)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:934)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1078)
        at java.base/java.lang.Thread.run(Thread.java:833)
```

In case it's helpful, Jetty is running behind a Cloudflare tunnel and seems to correlate with logs such as these slightly-redacted ones from `cloudflared`:

```
2023-01-16T20:40:26Z ERR  error="Incoming request ended abruptly: context canceled" cfRay=0000000000000000-LAX ingressRule=0 originService=https://jetty:8443
2023-01-16T20:40:26Z ERR Request failed error="Incoming request ended abruptly: context canceled" connIndex=0 dest=http://$DOMAIN/wp-includes/wlwmanifest.xml ip=198.41.192.167 type=http
```

It seems that the request ends abruptly according to `cloudflared`, and then Jetty tries to send a response, causing the job to fail, the exception to be thrown, and subsequently the log to show up.

**Question**

What is the correct way to prevent this from occurring?

**Comment by sbordet** [comment]
Seems to be a bad client that does not send the request properly.

This is probably coupled with a bad usage on the server of the `AsyncContext` timeout that fires and tries to write a response exactly a moment after a response has already been written.

**Comment by dutchmahoney** [comment]
Thanks for following up, @sbordet. Although I haven't been able to reproduce the issue by sending the request myself (though perhaps I should try a little harder :), I believe the request is being sent properly as I'm able to see the entirety of the request on the server side. To add a little more context, when I receive requests from scanners (any requests for PHP or XML in my case), I add some artificial delay by putting the requests on a `DelayQueue` to not block, and fulfill them after some period of time which does work. Although atypical, this seemed like it'd be a good test of Jetty's async capabilities by simulating a request that takes some time compute a result.

What is _not_ a "bad usage" of the `AsyncContext` timeout? IIUC, since any request in theory can be ended "abruptly", then there's no value that would prevent this from happening.

If my assumptions are correct, I also don't think there's a check that could be done prior to sending the response as that would simply introduce a race condition. With that being said, since this exception is unable to be handled by Jetty users, is this issue less a question and more a bug?

**Comment by sbordet** [comment]
In my experience, I never found the `AsyncContext` timeout useful, so I always disable it by setting it to `0`.
Failing to do that results in a timeout by default after 30 seconds, and likely to cause races with the idle timeout, or the load balancer idle timeout, or -- in your case -- the "cloudflare tunnel" timeout, or other things.

The stack trace you report shows that the `AsyncContext` timeout is expiring, but it is finding the response already written.
May happen due to request delays, or forgetting to call `AsyncContext.complete()`.

Are you really using the `AsyncContext` timeout for something useful?
If not, disable it.

Bottom line is that it's likely an application mistake.
